runsc/container/container.go

```diff
@@ -912,13 +929,25 @@ func (c *Container) forEachSelfMount(fn func(mountSrc string)) {
 // createGoferFilestores creates the regular files that will back the
 // tmpfs/overlayfs mounts that will overlay some gofer mounts. It also returns
 // information about how each gofer mount is configured.
-func (c *Container) createGoferFilestores(ovlConf config.Overlay2, mountHints *boot.PodMountHints) ([]*os.File, []boot.GoferMountConf, error) {
+func (c *Container) createGoferFilestores(ovlConf config.Overlay2, mountHints *boot.PodMountHints, rootfsHint *boot.RootfsHint) ([]*os.File, []boot.GoferMountConf, error) {
 	var goferFilestores []*os.File
 	var goferConfs []boot.GoferMountConf
 
 	// Handle root mount first.
-	shouldOverlay := ovlConf.RootEnabled() && !c.Spec.Root.Readonly
-	filestore, goferConf, err := c.createGoferFilestore(ovlConf, c.Spec.Root.Path, shouldOverlay, nil /* hint */)
+	overlayMedium := ovlConf.RootOverlayMedium()
+	mountType := boot.Bind
+	if rootfsHint != nil {
+		if !rootfsHint.Overlay.IsUnset() {
+			overlayMedium = rootfsHint.Overlay
+		}
```

Comment:
Hmm, so if overlay is not specified, the the overlay configuration from `--overlay2` will be used? That will be confusing.

I think we should either make the overlay annotation required, or provide a `none` default for it.
It is nice to have the zero value of a type be the default... I think both "" and "none" should map to `NoOverlay`. Then we don't have this problem.

Reply:
Makes sense. It is better to have the zero value be the default. And `NoOverlay` as the default is more digestible. If rootfs hint is provided by users, the default setting (i.e. `NoOverlay`) should be used when the overlay annotation is not specified, and when users want overlay, they can specify the corresponding overlay that they want. I will take this approach. Thanks!